code/harness/c_harness.cpp

```diff
@@ -139,7 +138,27 @@ int main(int argc, char** argv) {
    ****************************************************************************/
   const unsigned int test_case_idx = optind;
   const string path = argv[test_case_idx];
-   
+  //this should be changed in the option is added to mount tests in other directories
+  string mount_dir = "/mnt/snapshot"; 
+  if(setenv("MOUNT_FS", mount_dir.c_str(), 1) == -1){
+    cerr << "Error setting environment variable MOUNT_FS" << endl;
+  }
+  
+  //input file created to read output from bash command "df"
+  FILE *input;
+  char buf[512];
+  if(!(input = popen(("df --output=source,size | grep " + flags_dev).c_str(), "r"))){
```

Comment:
this will actually get the size of the `udev` file system which is different than the device size. You will likely need to substitute this with a call to `fdisk -l <test device path>`.

Reply:
Just kidding, it gets the size of the file system on the "flag device" (device we copy IO scheduler flags from), not the size of the test device.